x-pack/plugins/notifications/common/lib/promise_utils.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export type ResolvablePromise<T> = Promise<T> & {
+  doResolve: (value: unknown) => void;
+  doReject: (reason?: any) => void;
+};
+
+export function getResolvablePromise<T>(): ResolvablePromise<T> {
```

Comment:
nit: if depending on kibana_utils is not a problem, can user existing util: https://github.com/elastic/kibana/blob/main/src/plugins/kibana_utils/common/defer.ts

Reply:
N/A anymore, as the plugin contract is now synchronous.